src/components/modal/__internal__/modal-manager.js

```diff
@@ -1,26 +1,54 @@
 class ModalManagerInstance {
   #modalList = [];
 
-  addModal = (modal) => {
-    this.#modalList.push(modal);
+  #getTopModal() {
+    if (!this.#modalList.length) {
+      return {};
+    }
+    return this.#modalList[this.#modalList.length - 1];
+  }
+
+  addModal = (modal, setTriggerRefocusFlag) => {
+    const {
+      modal: topModal,
+      setTriggerRefocusFlag: setTrapFlag,
+    } = this.#getTopModal();
+
+    if (topModal && setTrapFlag) {
+      setTrapFlag(false);
+    }
+
+    this.#modalList.push({ modal, setTriggerRefocusFlag });
   };
 
   isTopmost(modal) {
-    if (!modal || !this.#modalList.length) {
+    if (!modal || !this.#getTopModal().modal) {
       return false;
     }
+    const { modal: topModal } = this.#getTopModal();
 
-    return this.#modalList.indexOf(modal) === this.#modalList.length - 1;
+    return modal === topModal;
   }
 
   removeModal(modal) {
-    const modalIndex = this.#modalList.indexOf(modal);
+    const modalIndex = this.#modalList.findIndex(({ modal: m }) => m === modal);
```

Comment:
Why did you change this?

Reply:
The list has been changed to contain object elements rather than just the refs so the old implementation won't work